syntax = "proto3";

package accounts;

service AccountService {
    rpc CreateAccount (CreateAccountRequest) returns (CreateAccountResponse);
}

message Account {
    int64 account_id = 1;
    string email = 2;
    string first_name = 4;
    string last_name = 5;
    bool is_active = 6;
    bool is_verified = 7;
}

message CreateAccountRequest {
    string email = 1;
    string first_name = 2;
    string last_name = 3;
    string hashed_password = 4;
}

message CreateAccountResponse {
    Account account = 1;
    string token = 2;
}
